packages/orchestrator/internal/template/build/commands/copy.go

```diff
@@ -164,15 +184,36 @@ func (c *Copy) Execute(
 	// Remove all glob patterns, they are handled on the client side already
 	// Add / always at the end to ensure the last file/directory is also included if it doesn't contain a glob pattern
 	sourcePath, _ := doublestar.SplitPattern(ensureTrailingSlash(args[0]))
+
+	// Determine the target owner
+	owner := fmt.Sprintf("%s:%s", cmdMetadata.User, cmdMetadata.User)
+	if len(args) >= 3 && args[2] != "" {
+		// Assumes the format of chown: user:group or just user
+		owner = args[2]
+		if !strings.Contains(owner, ":") {
+			owner = fmt.Sprintf("%s:%s", owner, owner)
+		}
+	}
+
+	// Get optional permissions
+	permissions := ""
+	if len(args) >= 4 {
+		permissions = args[3]
+	}
+
```

Comment:
Don't we want to extract this logic and unit test it? Can get really messy and confusing with more arguments coming.